examples/src/test/java/org/neo4j/docs/driver/ExamplesIT.java

```diff
@@ -563,14 +565,17 @@ void testShouldRunRxTransactionFunctionExampleReactor() throws Exception
             }
 
             StdIOCapture stdIOCapture = new StdIOCapture();
+            CountDownLatch latch = new CountDownLatch( 1 );
 
             // print all 'Product' nodes to fake stdout
             try ( AutoCloseable ignore = stdIOCapture.capture() )
             {
-                ResultSummary summary = await( Flux.from( example.printAllProductsReactor() ).single() );
-                assertEquals( StatementType.READ_ONLY, summary.statementType() );
+                Mono.from( example.printAllProductsReactor() )
+                        .doAfterTerminate( () -> latch.countDown() )
+                        .subscribe( summary -> assertEquals( StatementType.READ_ONLY, summary.statementType() ) );
             }
 
+            latch.await( 1, TimeUnit.MINUTES );
```

Comment:
I am not sure why this latch is needed? The original `await` has a timeout, which ensures the test will not blocking forever. If you wonder why the test is not written in a proper reactive tests it merely because this test's focus is not to verify reactive but the result is correct.

Another note: there is a difference between `Mono.from(manyPublisher)` and `Flux.from(manyPublisher).single()`. If the `manyPublisher` publishes more than one result, the first Mono code will success, the second Flux code will error.

Reply:
The latch has the same purpose as the `await`, but we cannot use `await` from `TestUtils` as they expose Reactor Types from the shaded package. Concrete: `org.neo4j.docs.driver.RxTransactionFunctionExample#printAllProductsReactor` returns `reactor.core.publisher` which now clashes with `org.neo4j.driver.internal.shaded.*`

The `await` on the latch has the same timeout.

Regarding Mono and Flux: You're right, I'll fix this.